src/swarm/neo/util/AcquiredResources.d

```diff
@@ -88,15 +100,17 @@ public struct AcquiredArraysOf ( T )
         }
 
         // Acquire container buffer, if not already done.
-        if ( this.acquired is null )
+        if ( this.buffer is null )
         {
-            this.acquired = cast(T[][])newBuffer((T[]).sizeof * 4);
+            this.buffer = newBuffer((void[]).sizeof * 4);
+            this.acquired = VoidBufferAsArrayOf!(void[])(&this.buffer);
         }
 
         // Acquire and re-initialise new buffer to return to the user. Store
         // it in the container buffer.
-        this.acquired ~= cast(T[])newBuffer(4);
-        return &this.acquired[$-1];
+        this.acquired ~= newBuffer(T.sizeof * 4);
+
+        return cast(T[]*)this.acquired.array()[this.acquired.length-1];
```

Comment:
```d
void[][] array_of_void_slices = this.acquired.array();
auto array_of_T_slices = cast(T[][])array_of_void_slices;
return &array_of_T_slices[$ - 1];
```
General advice to avoid corruption trouble: Don't cast pointers to arrays. Instead cast the array first, then use the `&` operator.